modules/src/ics04_channel/context.rs

```diff
@@ -37,9 +37,25 @@ pub trait ChannelReader {
 
     fn get_next_sequence_send(&self, port_channel_id: &(PortId, ChannelId)) -> Option<Sequence>;
 
+    fn get_next_sequence_recv(&self, port_channel_id: &(PortId, ChannelId)) -> Option<Sequence>;
```

Comment:
We don't have to do this in this PR, but I think that in general we should change the type of the key for the getters from:

```rust
key: &(A, B, C)
```

to

```rust
key: (&A, &B, &C)
```

to avoid having to `.clone()` the A, B and C at call site.

Reply:
Or maybe add a new type that encapsulates this tuple? We wanted to this originally but kept postponing it.